pip/index.py

```diff
@@ -153,6 +165,18 @@ def __init__(self, find_links, index_urls, allow_all_prereleases=False,
         # The Session we'll use to make requests
         self.session = session
 
+        # The valid tags to check potential found wheel candidates against
+        self.valid_tags = get_supported(
+            versions=versions,
+            platform=platform,
+            abi=abi,
+            impl=implementation,
+        )
+        self.valid_tags_noarch = get_supported(
```

Comment:
The `noarch` one?  I also don't see where that's used, removing it now.  `valid_tags` is used in several places.